tests/int_sat_test.go

```diff
@@ -0,0 +1,148 @@
+//go:build integration
+// +build integration
+
+package tests
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+	tc "github.com/testcontainers/testcontainers-go"
+)
+
+// IntegrationSuite will test @todo complete this.
+type IntegrationSuite struct {
+	suite.Suite
+
+	container tc.Container
+	ctxCloser context.CancelFunc
+}
+
+// TestIntegrationSuite gets run from go's test framework that kicks off the suite.
+func TestIntegrationSuite(t *testing.T) {
+	suite.Run(t, new(IntegrationSuite))
+}
+
+// SetupSuite runs first in the chain. Used to set up properties and settings
+// that all test methods need access to.
+func (i *IntegrationSuite) SetupSuite() {
+	dir, err := os.Getwd()
+	if err != nil {
+		i.FailNowf("getwd", "what: %s", err.Error())
+	}
+
+	satWorkingDir := filepath.Join(dir, "../examples")
+	ctx, ctxCloser := context.WithTimeout(context.Background(), 10*time.Second)
+	i.ctxCloser = ctxCloser
+
+	req := tc.GenericContainerRequest{
+		ContainerRequest: tc.ContainerRequest{
+			Image: "suborbital/sat:dev",
+			Env: map[string]string{
+				"SAT_HTTP_PORT": "8080",
+			},
+			ExposedPorts: []string{"8080:8080"},
+			Cmd: []string{
+				"sat", "/runnables/hello-echo/hello-echo.wasm",
+			},
+			BindMounts: map[string]string{
+				"/runnables": satWorkingDir,
+			},
+		},
+		Started:      true,
+		ProviderType: tc.ProviderDocker,
+	}
+
+	container, err := tc.GenericContainer(ctx, req)
+	if err != nil {
+		i.FailNowf("could not start up container", "error received: %s", err.Error())
+	}
+
+	i.container = container
+
+	i.T().Log("starting up docker...")
+	time.Sleep(2 * time.Second)
+}
+
+// TearDownSuite runs last, and is usually used to close database connections
+// or clear up after running the suite.
+func (i *IntegrationSuite) TearDownSuite() {
+	i.ctxCloser()
+	terminateCtx, termCtxCloser := context.WithTimeout(context.Background(), 3*time.Second)
+	defer termCtxCloser()
+	err := i.container.Terminate(terminateCtx)
+	if err != nil {
+		i.FailNowf("terminate failed", err.Error())
+	}
+	i.T().Log("hold on, terminating docker container...")
```

Comment:
Any reason not to use an Exit Strategy instead? 

https://pkg.go.dev/github.com/testcontainers/testcontainers-go@v0.12.0/wait#ExitStrategy

Reply:
Implemented in https://github.com/suborbital/sat/pull/65/commits/c1e898e8e3e345018bc1afb4f31168969c4a5df5